app-modules/service-management/src/Models/ProductLicense.php

```diff
@@ -0,0 +1,109 @@
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Models;
+
+use Carbon\Carbon;
+use App\Models\BaseModel;
+use AidingApp\Contact\Models\Contact;
+use OwenIt\Auditing\Contracts\Auditable;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Relations\BelongsTo;
+use Illuminate\Database\Eloquent\Attributes\ObservedBy;
+use AidingApp\Audit\Models\Concerns\Auditable as AuditableTrait;
+use AidingApp\ServiceManagement\Observers\ProductLicenseObserver;
+
+#[ObservedBy(ProductLicenseObserver::class)]
+
+class ProductLicense extends BaseModel implements Auditable
+{
+    use AuditableTrait;
+    use SoftDeletes;
+
+    protected $fillable = [
+        'license',
+        'description',
+        'start_date',
+        'expiration_date',
+        'additional_notes',
+    ];
+
+    protected $casts = [
+        'license' => 'encrypted',
+        'start_date' => 'date',
+        'expiration_date' => 'date',
+    ];
+
+    /**
+     * The accessors to append to the model's array form.
+     *
+     * @var array
+     */
+    protected $appends = ['status'];
+
+    public function product(): BelongsTo
+    {
+        return $this->belongsTo(Product::class, 'product_id', 'id');
+    }
+
+    public function contact(): BelongsTo
+    {
+        return $this->belongsTo(Contact::class, 'assigned_to', 'id');
+    }
+
+    protected function Status(): Attribute
+    {
+        return new Attribute(
+            get: function () {
+                $today = Carbon::today();
+
+                $startDate = $this->getAttribute('start_date');
+                $endDate = $this->getAttribute('end_date');
```

Comment:
Instead of calling `getAttribute` here I believe the closure should have the arguments $value and $attributes and you can pull from `$attributes`

https://laravel.com/docs/10.x/eloquent-mutators#building-value-objects-from-multiple-attributes